source/models/index.py

```diff
@@ -1,29 +1,33 @@
+from genericpath import isfile
 import os
 from outputs.exceptions import *
 
 class Index:
-    def __init__(self, table, index_value):
+    def __init__(self, table, index_name, index_value = None):
         self.table = table
+        self.index_name = index_name
         self.index_value = index_value
 
     def serialize(self):
         os.makedirs(self.get_path(), exist_ok = True)
-    
+
     def get_path(self, index_value = None):
         if index_value == None:
-            return os.path.join(self.table.get_path(), os.path.join("indices" , self.index_value))
+            return os.path.join(self.table.get_path(), os.path.join("Indices" , self.index_name))
         else:
             return os.path.join(self.get_path(), self.index_value + ".txt")
 
-    def add_primary_key(self, primary_key, index_value):
-        old_primary_keys = set(self.get_primary_key(index_value))                
+    def add_primary_key(self, primary_key):
```

Comment:
It should take index value.

index is index name and metadata that belong to.

Reply:
it take it from the self.
as I create the object I give it the ```index value``` and store it at the self